### PR TITLE
Optimize sort

### PR DESCRIPTION
**Pull request by ash2k**

### CODE REVIEW

File: pkg/ordering/sort.go [hunk 1]
```diff
@@ -61,58 +61,65 @@ func less(i, j object.ObjMetadata) bool {
...
+	orderLast = []string{
+		"ValidatingWebhookConfiguration",
+	}
+	kind2index = make(map[string]int, len(orderFirst)+len(orderLast))
+)
 
-// getIndexByKind returns the index of the kind respecting the order
-func getIndexByKind(kind string) int {
-	m := map[string]int{}
+func init() {
```
Comment (ash2k) [inline pkg/ordering/sort.go:97]: `init` precomputes the mapping.

File: pkg/ordering/sort.go [hunk 2]
```diff
@@ -61,58 +61,65 @@ func less(i, j object.ObjMetadata) bool {
...
 }
 
-func IsLessThan(x schema.GroupKind, o schema.GroupKind) bool {
+func IsLessThan(x, o schema.GroupKind) bool {
 	indexI := getIndexByKind(x.Kind)
 	indexJ := getIndexByKind(o.Kind)
 	if indexI != indexJ {
 		return indexI < indexJ
 	}
-	return x.String() < o.String()
```
Comment (ash2k) [inline pkg/ordering/sort.go:117]: Two allocations removed.

File: pkg/ordering/sort.go [hunk 3]
```diff
@@ -58,61 +58,70 @@ func less(i, j object.ObjMetadata) bool {
 	}
 	// In case of tie, compare the namespace and name combination so that the output
 	// order is consistent irrespective of input order
-	return i.Namespace+i.Name < j.Namespace+j.Name
```
Comment (ash2k) [inline pkg/ordering/sort.go:61]: Two allocations removed. Also, ns=`ab` + name=`c` vs ns=`a` + name=`bc` issue resolved.

File: pkg/ordering/sort.go [hunk 4]
```diff
@@ -58,61 +58,70 @@ func less(i, j object.ObjMetadata) bool {
...
+	}
+	return i.Name < j.Name
 }
 
-// getIndexByKind returns the index of the kind respecting the order
-func getIndexByKind(kind string) int {
-	m := map[string]int{}
+var kind2index map[string]int
+
+func init() {
```
Comment (ash2k) [inline pkg/ordering/sort.go:69]: `init` precomputes the mapping.

**Review by mortent — APPROVED** [review]
/lgtm
/approve

**Pull request merged**